src/ui/run-history-ui-handler.ts

```diff
@@ -0,0 +1,338 @@
+    this.runSelectMessageBoxContainer.setVisible(false);
+    this.runSelectContainer.add(this.runSelectMessageBoxContainer);
+
+    this.runSelectMessageBox = addWindow(this.scene, 1, -1, 318, 28);
+    this.runSelectMessageBox.setOrigin(0, 1);
+    this.runSelectMessageBoxContainer.add(this.runSelectMessageBox);
+
+    this.message = addTextObject(this.scene, 8, 8, "", TextStyle.WINDOW, { maxLines: 2 });
+    this.message.setOrigin(0, 0);
+    this.runSelectMessageBoxContainer.add(this.message);
+
+    this.runs = [];
+  }
+
+  show(args: any[]): boolean {
+    super.show(args);
+
+    this.getUi().bringToTop(this.runSelectContainer);
+    this.runSelectContainer.setVisible(true);
+    this.populateruns(this.scene);
+
+    this.setScrollCursor(0);
+    this.setCursor(0);
+
+    return true;
+  }
+
+  processInput(button: Button): boolean {
+    const ui = this.getUi();
+
+    let success = false;
+    const error = false;
+
+    if (button === Button.ACTION || button === Button.CANCEL) {
+      if (button === Button.ACTION) {
+        const cursor = this.cursor + this.scrollCursor;
+        if (this.runs[cursor]) {
+          console.log(this.runs[cursor].entryData);
+          this.scene.ui.setOverlayMode(Mode.RUN_INFO, this.runs[cursor].entryData, true);
+        } else {
+          return false;
+        }
+        success = true;
+        return success;
+      } else {
+        this.runSelectCallback = null;
+        success = true;
+        this.scene.ui.revertMode();
+      }
+    } else {
+      switch (button) {
+      case Button.UP:
+        if (this.cursor) {
+          success = this.setCursor(this.cursor - 1);
+        } else if (this.scrollCursor) {
+          success = this.setScrollCursor(this.scrollCursor - 1);
+        }
+        break;
+      case Button.DOWN:
+        if (this.cursor < 2) {
+          success = this.setCursor(this.cursor + 1);
+        } else if (this.scrollCursor < runCount - 3) {
+          success = this.setScrollCursor(this.scrollCursor + 1);
+        }
+        break;
+      }
+    }
+
+    if (success) {
+      ui.playSelect();
+    } else if (error) {
+      ui.playError();
+    }
+    return success || error;
+  }
+
+
+  async populateruns(scene: BattleScene) {
+    const response = await this.scene.gameData.getRunHistoryData(this.scene);
+    const timestamps = Object.keys(response);
+    const timestampsNo = timestamps.map(Number);
+    if (timestamps.length > 1) {
+      timestampsNo.sort((a, b) => a - b);
+    }
+    timestampsNo.reverse();
+    const entryCount = timestamps.length;
+    for (let s = 0; s < entryCount; s++) {
+      const entry = new RunEntry(this.scene, response, timestampsNo[s].toString(), s);
+      this.scene.add.existing(entry);
+      this.runsContainer.add(entry);
+      this.runs.push(entry);
+    }
+  }
+
+  showText(text: string, delay?: integer, callback?: Function, callbackDelay?: integer, prompt?: boolean, promptDelay?: integer) {
```

Comment:
no more `integer` please

```suggestion
  showText(text: string, delay?: number, callback?: Function, callbackDelay?: number, prompt?: boolean, promptDelay?: number) {
```